inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp

```diff
@@ -806,7 +806,7 @@ layout layout_optimizer::get_expected_layout(layout const& current_layout,
                                              layout const& output_or_weights_layout) {
     auto prim = node.get_primitive();
     auto expected_tensor = current_layout.size;
-    auto expected_data_type = data_types::f32;
+    auto expected_data_type = current_layout.data_type;
```

Comment:
Have you checked that this change doesn't impact the accuracy metrics for fp16?

Reply:
Yes, I checked there was accuracy issue for fp16 model. So I've reverted layout_optimizer and reorder_inputs code change to the previous one.